task_sdk/src/airflow/sdk/definitions/asset/__init__.py

```diff
@@ -54,9 +55,26 @@ class AssetUniqueKey(NamedTuple):
     uri: str
 
     @staticmethod
-    def from_asset(asset: Asset) -> AssetUniqueKey:
+    def from_asset(asset: Asset | AssetModel) -> AssetUniqueKey:
         return AssetUniqueKey(name=asset.name, uri=asset.uri)
 
+    def to_obj(self) -> Asset:
```

Comment:
I wonder if there’s a better name than `obj`… Python doesn’t seem to have an established convention for this. (Rust has `into()`, for example.)

Also maybe the `from` part should also be `from_obj`? Not sure.

Reply:
Or maybe we should just call these `to_asset` and `to_asset_alias`. We really only need polymorphism in one place (the outlet event accessor); it can be pretty easily avoided.

Reply:
> Or maybe we should just call these to_asset and to_asset_alias. We really only need polymorphism in one place (the outlet event accessor); it can be pretty easily avoided with an if-else condition.

Yep, it would work and that's the main reason I name it this way

Reply:
fixed